include/rocksdb/sst_partitioner.h

```diff
@@ -62,6 +62,13 @@ class SstPartitioner {
   virtual bool CanDoTrivialMove(const Slice& smallest_user_key,
                                 const Slice& largest_user_key) = 0;
 
+  struct Segment {
```

Comment:
This is cumbersome to implement in ffi. You should use two separate vectors instead.